packages/backend/src/models/CommentModel/CommentModel.ts

```diff
@@ -0,0 +1,229 @@
+            )
+            .leftJoin(
+                'saved_queries',
+                'saved_queries.saved_query_id',
+                'dashboard_tile_charts.saved_chart_id',
+            )
+            .where(`${DashboardsTableName}.dashboard_uuid`, dashboardUuid)
+            .andWhere(
+                `${DashboardTilesTableName}.dashboard_tile_uuid`,
+                dashboardTileUuid,
+            )
+            .select(
+                `${DashboardTilesTableName}.dashboard_tile_uuid`,
+                `saved_queries.saved_query_uuid`,
+            )
+            .first();
+
+        if (!dashboardTile) {
+            throw new NotFoundError('Dashboard tile not found for dashboard');
+        }
+        return { savedChartUuid: dashboardTile.saved_query_uuid };
+    }
+
+    async createComment(
+        dashboardUuid: string,
+        dashboardTileUuid: string,
+        text: string,
+        replyTo: string | null,
+        user: LightdashUser,
+    ): Promise<string> {
+        const { savedChartUuid } =
+            await this.checkDashboardTileExistsInDashboard(
+                dashboardUuid,
+                dashboardTileUuid,
+            );
+        const [{ comment_id: commentId }] = await this.database(
+            DashboardTileCommentsTableName,
+        )
+            .insert({
+                text,
+                dashboard_tile_uuid: dashboardTileUuid,
+                reply_to: replyTo ?? null,
+                user_uuid: user.userUuid,
+                saved_chart_uuid: savedChartUuid,
+            })
+            .returning('comment_id');
+
+        return commentId;
+    }
+
+    async findCommentsForDashboard(
+        dashboardUuid: string,
+        userUuid: string,
+        canUserRemoveAnyComment: boolean,
+    ): Promise<Record<string, Comment[]>> {
+        const dashboard = await this.database(DashboardsTableName)
+            .leftJoin(
+                DashboardVersionsTableName,
+                `${DashboardsTableName}.dashboard_id`,
+                `${DashboardVersionsTableName}.dashboard_id`,
+            )
+            .where('dashboard_uuid', dashboardUuid)
+            .orderBy(`${DashboardVersionsTableName}.created_at`, 'desc')
+            .limit(1)
+            .first();
+
+        if (dashboard === undefined)
+            throw new NotFoundError('Dashboard not found');
+
+        const tiles = await this.database(DashboardTilesTableName)
+            .select('dashboard_tile_uuid')
+            .where('dashboard_version_id', dashboard.dashboard_version_id);
+        const tileUuids = tiles.map((tile) => tile.dashboard_tile_uuid);
+
+        const commentsWithUsers = await this.database(
+            DashboardTileCommentsTableName,
+        )
+            .leftJoin(
+                UserTableName,
+                `${DashboardTileCommentsTableName}.user_uuid`,
+                '=',
+                `${UserTableName}.user_uuid`,
+            )
+            .select(
```

Comment:
.select<
                (DbDashboardTileComments &
                    Pick<DbUser, 'first_name' | 'last_name'>)[]
            >(
```suggestion
            .select(
```